File: backend/app/schemas/review.py
"""人工审核相关数据模型。"""

from typing import Any

from pydantic import BaseModel, Field


class IdentityReviewItem(BaseModel):
    source_person_id: int
    source_name: str | None = None
    target_person_id: int
    target_name: str | None = None
    confidence: float = 0
    reason: str = ""
    evidence: dict[str, Any] | None = None
    source_passages: list[str] = Field(default_factory=list)
    target_passages: list[str] = Field(default_factory=list)
    hops: int | None = None
    llm_decision: str | None = None
    llm_used_full_text: bool = False
    annotated: bool = False


class IdentityReviewListResponse(BaseModel):
    pending_count: int
    items: list[IdentityReviewItem]


class PassageText(BaseModel):
    doc_id: int
    title: str
    context: str


class IdentityDecisionLog(BaseModel):
    """LLM 逐轮同名判断留痕（来自 identity_resolution_decision_logs）。"""

    hop: int
    decision: str
    confidence: float = 0
    reason: str = ""
    positive_evidence: list[str] = Field(default_factory=list)
    negative_evidence: list[str] = Field(default_factory=list)
    missing_evidence: list[str] = Field(default_factory=list)
    used_full_text: bool = False
    created_at: str | None = None


class IdentityAIReportRead(BaseModel):
    id: int
    source_person_id: int
    target_person_id: int
    source_name: str | None = None
    target_name: str | None = None
    report_markdown: str
    updated_at: str | None = None


class IdentityEvidenceResponse(BaseModel):
    source_evidence: dict[str, Any]
    target_evidence: dict[str, Any]
    graph_elements: dict[str, Any]
    review_relation: dict[str, Any] | None = None
    decision_logs: list[IdentityDecisionLog] = Field(default_factory=list)
    source_passage_texts: list[PassageText] = Field(default_factory=list)
    target_passage_texts: list[PassageText] = Field(default_factory=list)
    ai_report: IdentityAIReportRead | None = None


class GenerateIdentityReportsRequest(BaseModel):
    force: bool = False


class GenerateIdentityReportsResponse(BaseModel):
    pair_count: int
    created_count: int
    updated_count: int
    skipped_count: int
    failed_count: int
    failures: list[dict[str, Any]] = Field(default_factory=list)


class AdjudicateRequest(BaseModel):
    decision: str  # "merge" or "keep_separate"


class AdjudicateResponse(BaseModel):
    status: str
    action: str
    detail: dict[str, Any] = Field(default_factory=dict)


class AnnotateRequest(BaseModel):
    # 置信度本身即方向信号（0=非常确定不是同人，10=非常确定是同人），
    # decision 可省略，由后端按置信度推导，仅为兼容旧数据保留。
    human_confidence: int = Field(ge=0, le=10)
    decision: str | None = None
    note: str = ""


class AnnotateResponse(BaseModel):
    status: str
    annotation_id: int

File: frontend/src/api/review.ts
import apiClient from './client'

export interface IdentityReviewItem {
  source_person_id: number
  source_name: string | null
  target_person_id: number
  target_name: string | null
  confidence: number
  reason: string
  evidence: Record<string, unknown> | null
  source_passages: string[]
  target_passages: string[]
  hops: number | null
  llm_decision: string | null
  llm_used_full_text: boolean
  annotated: boolean
}

export interface IdentityAIReport {
  id: number
  source_person_id: number
  target_person_id: number
  source_name: string | null
  target_name: string | null
  report_markdown: string
  updated_at: string | null
}

export interface IdentityReviewListResponse {
  pending_count: number
  items: IdentityReviewItem[]
}

export interface GraphElements {
  nodes: Array<{ id: string; label: string; type: string; properties: Record<string, unknown> }>
  edges: Array<{ source: string; target: string; label: string; properties: Record<string, unknown> }>
}

export interface PassageText {
  doc_id: number
  title: string
  context: string
}

export interface IdentityDecisionLog {
  hop: number
  decision: string
  confidence: number
  reason: string
  positive_evidence: string[]
  negative_evidence: string[]
  missing_evidence: string[]
  used_full_text: boolean
  created_at: string | null
}

export interface IdentityEvidenceResponse {
  source_evidence: Record<string, unknown>
  target_evidence: Record<string, unknown>
  graph_elements: GraphElements
  review_relation: Record<string, unknown> | null
  decision_logs: IdentityDecisionLog[]
  source_passage_texts: PassageText[]
  target_passage_texts: PassageText[]
  ai_report: IdentityAIReport | null
}

export interface AdjudicateResponse {
  status: string
  action: string
  detail: Record<string, unknown>
}

export interface AnnotateResponse {
  status: string
  annotation_id: number
}

export interface GenerateIdentityReportsResponse {
  pair_count: number
  created_count: number
  updated_count: number
  skipped_count: number
  failed_count: number
  failures: Array<Record<string, unknown>>
}

export async function listPendingReviews(mode: 'pending' | 'annotation' = 'pending') {
  const { data } = await apiClient.get<IdentityReviewListResponse>(
    '/review/identity-pending',
    { params: { mode } },
  )
  return data
}

export async function getReviewEvidence(sourceId: number, targetId: number) {
  const { data } = await apiClient.get<IdentityEvidenceResponse>(
    `/review/identity/${sourceId}/${targetId}/evidence`,
  )
  return data
}

export async function adjudicateIdentity(sourceId: number, targetId: number, decision: string) {
  const { data } = await apiClient.post<AdjudicateResponse>(
    `/review/identity/${sourceId}/${targetId}/adjudicate`,
    { decision },
  )
  return data
}

export async function annotateIdentity(
  sourceId: number,
  targetId: number,
  humanConfidence: number,
  note: string = '',
) {
  const { data } = await apiClient.post<AnnotateResponse>(
    `/review/identity/${sourceId}/${targetId}/annotate`,
    { human_confidence: humanConfidence, note },
  )
  return data
}

export async function generateIdentityReports(force = false) {
  const { data } = await apiClient.post<GenerateIdentityReportsResponse>(
    '/review/identity/reports/generate',
    { force },
  )
  return data
}
